src/Kinto.elm

```diff
@@ -137,23 +147,35 @@ endpointUrl config endpoint =
                 joinUrl [ baseUrl, "buckets", bucketName, "collections", collectionName, "records", recordId ]
 
 
-kintoRequest : Config -> Endpoint -> Method -> Http.Request Json.Value
-kintoRequest config endpoint method =
-    Http.request
-        { method = method
-        , headers = headersFromConfig config
-        , url = endpointUrl config endpoint
-        , body = Http.emptyBody
-        , expect = Http.expectJson bodyDataDecoder
-        , timeout = Nothing
-        , withCredentials = False
-        }
-
-
-performQuery : Config -> Endpoint -> Method -> (Result Error Json.Value -> msg) -> Cmd msg
-performQuery config endpoint method toMsg =
-    kintoRequest config endpoint method
-        |> Http.send (toKintoResponse >> toMsg)
+fromRequest : Request -> Http.Request Encode.Value
+fromRequest request =
+    case request of
+        KintoRequest config endpoint method ->
+            Http.request
+                { method = method
+                , headers = headersFromConfig config
+                , url = endpointUrl config endpoint
+                , body = Http.emptyBody
+                , expect = Http.expectJson bodyDataDecoder
+                , timeout = Nothing
+                , withCredentials = False
+                }
+
+        KintoRequestWithBody config endpoint method body ->
+            Http.request
+                { method = method
+                , headers = headersFromConfig config
+                , url = endpointUrl config endpoint
+                , body = Http.jsonBody body
+                , expect = Http.expectJson bodyDataDecoder
+                , timeout = Nothing
+                , withCredentials = False
+                }
```

Comment:
Lots of duplication here! I think we could at least factor out a `commonBody`, right? Or would that be overly complicated?